### ISSUE TITLE
Won't Allow "Size" as Custom Product Attribute for Variable Product

### ISSUE THREAD

**Issue by elysiumonline** [woocommerce/woocommerce#14844]
### EXPLANATION OF THE ISSUE

When trying to add sizing variations via the "Custom Product Attribute" option under "Attributes" of Variable Product as title "Size", attribute will not save. When trying to duplicate existing product with "Size" custom attribute (that was made without problems before update) the "Size" attribute disappears. I can seemingly name the custom product attribute anything other than "Size" and it will show up, it is only "Size" that Woocommerce has a problem with.

Anyone else experiencing this?

**Only fixed this issue when downgrading to Woocommerce 3.0.4

### STEPS TO REPRODUCE THE ISSUE

1. Create new product
2. Specify "Variable Product"
3. Try to add Custom Product Attribute "Size" under "Attributes" tab.
4. Save product Draft
5. Check attributes or variations to see if "Size" is still there.

### SYSTEM STATUS

<details>
```
WordPress environment
Home URL:		http://elysiumstore.online
Site URL:		http://elysiumstore.online
WC version:		3.0.5
Log directory writable:		 /home2/wael760acwg6/public_html/wp-content/uploads/wc-logs/
WP version:		4.7.4
WP multisite:		–
WP memory limit:		256 MB
WP debug mode:		–
WP cron:		
Language:		en_US
Server environment
Server info:		Apache
PHP version:		5.6.30
PHP post max size:		64 MB
PHP time limit:		30
PHP max input vars:		1000
cURL version:		7.19.7, NSS/3.27.1
SUHOSIN installed:		–
MySQL version:		 5.5.51 - We recommend a minimum MySQL version of 5.6. See: WordPress requirements
Max upload size:		64 MB
Default timezone is UTC:		
fsockopen/cURL:		
SoapClient:		
DOMDocument:		
GZip:		
Multibyte string:		
Remote post:		
Remote get:		
Database
WC database version:		3.0.5
Database prefix	 	wp_
woocommerce_sessions	 	
woocommerce_api_keys	 	
woocommerce_attribute_taxonomies	 	
woocommerce_downloadable_product_permissions	 	
woocommerce_order_items	 	
woocommerce_order_itemmeta	 	
woocommerce_tax_rates	 	
woocommerce_tax_rate_locations	 	
woocommerce_shipping_zones	 	
woocommerce_shipping_zone_locations	 	
woocommerce_shipping_zone_methods	 	
woocommerce_payment_tokens	 	
woocommerce_payment_tokenmeta	 	
MaxMind GeoIP database:		 The MaxMind GeoIP Database does not exist - Geolocation will not function. You can download and install it manually from http://dev.maxmind.com/geoip/legacy/geolite/ to the path: /home2/wael760acwg6/public_html/wp-content/uploads/GeoIP.dat. Scroll down to "Downloads" and download the "Binary / gzip" file next to "GeoLite Country". Please remember to uncompress GeoIP.dat.gz and upload the GeoIP.dat file only.
Security
Secure connection (HTTPS):		Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors		
Active plugins (7)
Elegant Themes Support	 	by Elegant Themes – 1.2
Google Analytics Dashboard for WP	 	by Alin Marcu – 4.9.6.2
Regenerate Thumbnails	 	by Alex Mills (Viper007Bond) – 2.2.6
Velvet Blues Update URLs	 	by VelvetBlues.com – 3.2.7
WooCommerce Stripe Gateway	 	by WooCommerce – 3.1.7
WooCommerce	 	by Automattic – 3.0.5
WP Super Cache	 	by Automattic – 1.4.9
Settings
API enabled:		
Force SSL:		
Currency		USD ($)
Currency position		left
Thousand separator		,
Decimal separator		.
Number of decimals		0
Taxonomies: Product types		external (external), grouped (grouped), simple (simple), variable (variable)
Taxonomies: Product visibility		exclude-from-catalog (exclude-from-catalog), exclude-from-search (exclude-from-search), featured (featured), outofstock (outofstock), rated-1 (rated-1), rated-2 (rated-2), rated-3 (rated-3), rated-4 (rated-4), rated-5 (rated-5)
WC pages
Shop base:		#4 - /shop/
Cart:		#5 - /cart/
Checkout:		#6 - /cart/checkout/
My account:		#7 - /my-account/
Theme
Name:		Divi
Version:		3.0.42
Author URL:		http://www.elegantthemes.com
Child theme:		 – If you're modifying WooCommerce on a parent theme you didn't build personally, then we recommend using a child theme. See: How to create a child theme
WooCommerce support:		
```
</details>

**Comment by faisal-alvi** [comment]
I found this bug if I do not enter **Value(s):** for the "Size" attribute and click **Save Draft**.

But if I enter values, and then click **Save Draft** it saves it properly.

**Comment by warezit** [comment]
I have this exact same issue! Same for me... downgrading from 3.0.5 to 3.0.4 was my only workaround so far.

I started something about it on stackoverflow here: http://stackoverflow.com/questions/43724547/woocommerce-attributes-not-showing-on-front-end

**Comment by claudiulodro** [comment]
I think this may be related to #14824. Do you have a global attribute named "Size" at the Products > Attributes screen.

**Comment by claudiosanches** [comment]
@faisal-alvi is correct, need to have at least one value before try to save.

**Comment by warezit** [comment]
To all: in my situation, I had already setup the "Size" attribute, with options, months ago. All of the products with the "size" variable have been working fine, since I set them up, until 3.0.5 was released.

Also, yes, in my case: I do have an Attribute named "Size" in the Global Attributes - under WooCommerce > Attributes. I also have custom attributes set for some products, also with the name "Size".

Hope this info helps somehow! I am happy to provide any other info!
